src/app/main.cpp

```diff
@@ -810,24 +817,52 @@ int main( int argc, char *argv[] )
   QCoreApplication::setApplicationName( QgsApplication::QGIS_APPLICATION_NAME );
   QCoreApplication::setAttribute( Qt::AA_DontShowIconsInMenus, false );
 
-  QSettings* customizationsettings = nullptr;
+  // SetUp the QgsSettings Global Settings:
+  // - use the path specified with --globalsettings path,
+  // - use the environment if not found
+  // - use a default location as a fallback
+  if ( globalsettingsfile.isEmpty( ) )
+  {
+    globalsettingsfile = getenv( "QGIS_GLOBAL_SETTINGS_FILE" );
+  }
+  if ( globalsettingsfile.isEmpty( ) )
+  {
+    QString default_globalsettingsfile = QgsApplication::pkgDataPath( ) + "/qgis_global_settings.ini";
```

Comment:
I might change this if this is ok based on the QEP.  It will just look in a `QgsApplication::pkgDataPath()/gloabl/QGIS/QGIS.ini` file.  As per the QEP just makes it easier and allows use to include other things in the "global config" as it's just a copy of a user config but in the install folder.  e.g we might want to include extra plugins, symbols, etc in there so we can setup a user config and then copy to install folder.